GradeCenter.Data/Models/School.cs

```diff
@@ -0,0 +1,28 @@
+﻿using GradeCenter.Data.Models.Common;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace GradeCenter.Data.Models
+{
+    public class School: BaseModel<int>
+    {
+        public School()
+        {
+            this.Students = new HashSet<Student>();
+            this.Parents = new HashSet<Parent>();
+        }
+
+        [Required]
+        public string Name { get; set; }
+
+        [Required]
+        public string Address { get; set; }
+
+        [Required]
+        [ForeignKey("Principal")]
+        public int PrincipalId { get; set; }
+        public Principal Principal { get; set; }
+        public ICollection<Student> Students { get; set; }
+        public ICollection<Parent> Parents { get; set; }
```

Comment:
Here you do not need the Parents collection as we will have access to them through Students. What you will need instead is a collection for Teachers. :)